airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/split_secrets/SecretsHelpers.java

```diff
@@ -0,0 +1,421 @@
+import io.airbyte.protocol.models.ConnectorSpecification;
+import io.airbyte.validation.json.JsonSchemaValidator;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+
+/**
+ * Contains most of the core logic surrounding secret coordinate extraction and insertion.
+ *
+ * These are the three main helpers provided by this class:
+ * {@link SecretsHelpers#splitConfig(UUID, JsonNode, ConnectorSpecification)}
+ * {@link SecretsHelpers#splitAndUpdateConfig(UUID, JsonNode, JsonNode, ConnectorSpecification, ReadOnlySecretPersistence)}
+ * {@link SecretsHelpers#combineConfig(JsonNode, SecretPersistence)}
+ *
+ * Here's an overview on some terminology used in this class:
+ *
+ * A "full config" represents an entire connector config as specified by an end user. This should
+ * conform to a connector specification.
+ *
+ * A "partial config" represents a connector config where any string that was specified as an
+ * airbyte_secret in the connector specification is replaced by a JSON object {"_secret": "secret
+ * coordinate"} that can later be used to reconstruct the "full config".
+ *
+ * A {@link SecretPersistence} provides the ability to read and write secrets to a backing store
+ * such as Google Secrets Manager.
+ *
+ * A {@link SecretCoordinate} is a reference to a specific secret at a specific version in a
+ * {@link SecretPersistence}.
+ */
+public class SecretsHelpers {
+
+  public static final String COORDINATE_FIELD = "_secret";
+
+  /**
+   * Used to separate secrets out of a connector configuration. This will output a partial config that
+   * includes pointers to secrets instead of actual secret values and a map that can be used to update
+   * a {@link SecretPersistence} at coordinates with values from the full config.
+   *
+   * @param workspaceId workspace used for this connector config
+   * @param fullConfig config including secrets
+   * @param spec connector specification
+   * @return a partial config + a map of coordinates to secret payloads
+   */
+  public static SplitSecretConfig splitConfig(final UUID workspaceId,
+                                              final JsonNode fullConfig,
+                                              final ConnectorSpecification spec) {
+    return internalSplitAndUpdateConfig(UUID::randomUUID, workspaceId, (coordinate) -> Optional.empty(), Jsons.emptyObject(), fullConfig,
+        spec.getConnectionSpecification());
+  }
+
+  /**
+   * Used to separate secrets out of a connector configuration and output a partial config that
+   * includes pointers to secrets instead of actual secret values and a map that can be used to update
+   * a {@link SecretPersistence} at coordinates with values from the full config. If the previous
+   * config for this connector's configuration, this method attempts to use the same base coordinate
```

Comment:
I think a word is missing here.  "If the previous config for this connector's configure [does what?], then..."